
# HG changeset patch
# User Paul Turner <paul@cegui.org.uk>
# Date 1303380086 -3600
# Node ID c813e7a3e87311cf52a4534fdd373e5d91ca8ee3
# Parent  a208cc7c172bc454fd41d13634fd6d57cb5516b3
FIX: include of cstddef header was missing for ptrdiff_t useage.  see: http://www.cegui.org.uk/phpBB2/viewtopic.php?f=3&t=5546

diff -r a208cc7c172b -r c813e7a3e873 cegui/include/CEGUIString.h
--- a/cegui/include/CEGUIString.h	Thu Apr 21 08:21:25 2011 +0100
+++ b/cegui/include/CEGUIString.h	Thu Apr 21 11:01:26 2011 +0100
@@ -31,9 +31,9 @@
 #define _CEGUIString_h_
 
 #include "CEGUIBase.h"
-#include <string>
 #include <string.h>
 #include <stdexcept>
+#include <cstddef>
 
 // Start of CEGUI namespace section
 namespace CEGUI
@@ -62,7 +62,7 @@
 	*************************************************************************/
 	typedef		utf32			value_type;					//!< Basic 'code point' type used for String (utf32)
 	typedef		size_t			size_type;					//!< Unsigned type used for size values and indices
-	typedef		ptrdiff_t		difference_type;			//!< Signed type used for differences
+	typedef		std::ptrdiff_t	difference_type;			//!< Signed type used for differences
 	typedef		utf32&			reference;					//!< Type used for utf32 code point references
 	typedef		const utf32&	const_reference;			//!< Type used for constant utf32 code point references
 	typedef		utf32*			pointer;					//!< Type used for utf32 code point pointers

